common.gypi

```diff
@@ -1,17 +1,7 @@
 {
   'target_defaults': {
     'default_configuration': 'Release',
-    'cflags_cc' : [
-      '-std=c++14',
-      # The assumption is that projects based on node-cpp-skel will also
-      # depend on mason packages. Currently (this will change in future mason versions)
-      # mason packages default to being built/linked with the CXX11_ABI=0.
-      # So we need to link this way too. This allows source
-      # compiling your module on any ubuntu version, even the latest
-      # where the CXX11_ABI default has flipped to 1
-      # More details at https://github.com/mapbox/mason/issues/319
-      '-D_GLIBCXX_USE_CXX11_ABI=0'
```

Comment:
Can you clarify why this is no longer needed? It also probably deserves a mention in the PR (and possibly changelog?)

Reply:
Initially I removed this due some build issue when using node 16. I remember the issue at that point was that this flag was duplicated in the build commands many times, few times with value 1 and once with value 0. I removed this line and it was not set as 0 anymore and build succeeded.

Now, even if I add this flag, it will still be 1 when running `make`
```
  c++ -o Release/obj.target/vtcomposite/src/module.o ../src/module.cpp '-DNODE_GYP_MODULE_NAME=vtcomposite' '-DUSING_UV_SHARED=1' '-DUSING_V8_SHARED=1' '-DV8_DEPRECATION_WARNINGS=1' '-DV8_DEPRECATION_WARNINGS' '-DV8_IMMINENT_DEPRECATION_WARNINGS' '-D_GLIBCXX_USE_CXX11_ABI=1' '-D_DARWIN_USE_64_BIT_INODE=1' '-D_LARGEFILE_SOURCE' '-D_FILE_OFFSET_BITS=64' '-DOPENSSL_NO_PINSHARED' '-DOPENSSL_THREADS' '-DBUILDING_NODE_EXTENSION' '-DNDEBUG' -I/Users/oskari/Library/Caches/node-gyp/16.14.0/include/node -I/Users/oskari/Library/Caches/node-gyp/16.14.0/src -I/Users/oskari/Library/Caches/node-gyp/16.14.0/deps/openssl/config -I/Users/oskari/Library/Caches/node-gyp/16.14.0/deps/openssl/openssl/include -I/Users/oskari/Library/Caches/node-gyp/16.14.0/deps/uv/include -I/Users/oskari/Library/Caches/node-gyp/16.14.0/deps/zlib -I/Users/oskari/Library/Caches/node-gyp/16.14.0/deps/v8/include  -O3 -mmacosx-version-min=10.8 -arch x86_64 -Wall -Wendif-labels -W -Wno-unused-parameter -std=c++14 -stdlib=libc++ -fvisibility-inlines-hidden -isystem /Users/oskari/Projects/vtcomposite/node_modules/node-addon-api -isystem /Users/oskari/Projects/vtcomposite/mason_packages/.link/include/ -Wall -Wextra -Weffc++ -Wconversion -pedantic-errors -Wconversion -Wshadow -Wfloat-equal -Wuninitialized -Wunreachable-code -Wold-style-cast -Wno-error=unused-variable -Werror -flto -flto -MMD -MF ./Release/.deps/Release/obj.target/vtcomposite/src/module.o.d.raw -I/opt/homebrew/opt/ruby/include  -c
  c++ -o Release/obj.target/vtcomposite/src/vtcomposite.o ../src/vtcomposite.cpp '-DNODE_GYP_MODULE_NAME=vtcomposite' '-DUSING_UV_SHARED=1' '-DUSING_V8_SHARED=1' '-DV8_DEPRECATION_WARNINGS=1' '-DV8_DEPRECATION_WARNINGS' '-DV8_IMMINENT_DEPRECATION_WARNINGS' '-D_GLIBCXX_USE_CXX11_ABI=1' '-D_DARWIN_USE_64_BIT_INODE=1' '-D_LARGEFILE_SOURCE' '-D_FILE_OFFSET_BITS=64' '-DOPENSSL_NO_PINSHARED' '-DOPENSSL_THREADS' '-DBUILDING_NODE_EXTENSION' '-DNDEBUG' -I/Users/oskari/Library/Caches/node-gyp/16.14.0/include/node -I/Users/oskari/Library/Caches/node-gyp/16.14.0/src -I/Users/oskari/Library/Caches/node-gyp/16.14.0/deps/openssl/config -I/Users/oskari/Library/Caches/node-gyp/16.14.0/deps/openssl/openssl/include -I/Users/oskari/Library/Caches/node-gyp/16.14.0/deps/uv/include -I/Users/oskari/Library/Caches/node-gyp/16.14.0/deps/zlib -I/Users/oskari/Library/Caches/node-gyp/16.14.0/deps/v8/include  -O3 -mmacosx-version-min=10.8 -arch x86_64 -Wall -Wendif-labels -W -Wno-unused-parameter -std=c++14 -stdlib=libc++ -fvisibility-inlines-hidden -isystem /Users/oskari/Projects/vtcomposite/node_modules/node-addon-api -isystem /Users/oskari/Projects/vtcomposite/mason_packages/.link/include/ -Wall -Wextra -Weffc++ -Wconversion -pedantic-errors -Wconversion -Wshadow -Wfloat-equal -Wuninitialized -Wunreachable-code -Wold-style-cast -Wno-error=unused-variable -Werror -flto -flto -MMD -MF ./Release/.deps/Release/obj.target/vtcomposite/src/vtcomposite.o.d.raw -I/opt/homebrew/opt/ruby/include  -c
  c++ -bundle -Wl,-bind_at_load -undefined dynamic_lookup -Wl,-search_paths_first -flto -flto -Wl,-dead_strip -mmacosx-version-min=10.8 -arch x86_64 -L./Release -stdlib=libc++ -L/opt/homebrew/opt/ruby/lib -o /Users/oskari/Projects/vtcomposite/lib/binding/vtcomposite.node Release/obj.target/vtcomposite/src/module.o Release/obj.target/vtcomposite/src/vtcomposite.o
  ln -f "/Users/oskari/Projects/vtcomposite/lib/binding/vtcomposite.node" "Release/vtcomposite.node" 2>/dev/null || (rm -rf "Release/vtcomposite.node" && cp -af "/Users/oskari/Projects/vtcomposite/lib/binding/vtcomposite.node" "Release/vtcomposite.node")
run 'make clean' for full rebuild
```
So for that reason I think this is not necessary anymore with current versions of used libraries and node. I tried to read about the need for this flag from previous issues, and got an understanding that the flag was necessary to set as 0 to get the build succeed, which is not the case anymore --> remove the flag totally.

I'm not totally comfortable with c++ and these build related stuff, so if you know better or understand the situation better, I'd be glad to get tips on this :)

Reply:
From the later discussions, if you build multiple components with mason, you should have same build flag value for all of these. That's the case for example in vtshaver, which uses mbgl. 

vtcomposite does not use mbgl or others, so for this library this is not needed.

Reply:
John's comments elsewhere:
_this flag is not used by clang++ by default (AFAICT) so it wouldn't affect anything
clang++ uses their own standard library implementation (libc++) not the GCC one (libstdc++) https://clang.llvm.org/docs/Toolchain.html#id6 and this flag is supported by libstdc++ [gcc ref](https://gcc.gnu.org/onlinedocs/gcc-11.2.0/libstdc++/manual/manual/using_dual_abi.html)_